docs/agents/magpie.rst

```diff
@@ -0,0 +1,343 @@
+.. highlight:: rst
+
+.. _magpie:
+
+===============
+Magpie Agent
+===============
+
+The magpie is an incredibly intelligent bird, with a decent ability to mimic
+other bird calls, though not as good as `the superb lyrebird
+<https://www.youtube.com/watch?v=mSB71jNq-yQ>`_. In the context of OCS, the job
+of the Magpie agent is to take detector data from the SMuRF Streamer and
+translate into G3Frames that are compatible with lyrebird. This requires
+a small bit of pre-processing and downsampling that is done whenever a new
+frame is received. 
+
+.. argparse::
+   :filename: ../agents/magpie/magpie_agent.py
+   :func: make_parser
+   :prog: python3 magpie_agent.py
+
+Configuration File Examples
+------------------------------
+
+Below are example docker-compose and ocs configuration files for running the
+magpie agent.
+
+OCS Site Config
+``````````````````
+Below is the site-config entry for the magpie instance we have running at UCSD
+on K2SO. We are displaying detectors using a wafer layout, determined by
+a detmap csv file, with a target sample rate of 20 Hz::
+
+      {'agent-class': 'MagpieAgent',
+       'instance-id': 'magpie-crate1slot2',
+       'arguments': [
+         '--stream-id', 'crate1-slot2',
+
+         '--src', 'tcp://localhost:4532',
+
+         '--dest', 8675,
+         '--delay', 5,
+         '--target-rate', 20,
+
+         '--layout', 'wafer',
+         # Detmap CSV file
+         '--det-map', '/home/jlashner/lyrebird_demo/detmap.csv',
+         '--offset', 0, 0,
+       ]},
+
+It is  possible to tell magpie to stream data from existing G3Files instead of
+directly from the smurf-stream. To do this, simply set the ``src`` argument to
+be the filepath of the file you wish to stream. If you want to stream from
+multiple files in series, you can do this by putting multiple source filepaths,
+as is shown below. In this case, once the first file has finished streaming
+magpie will imediately start streaming from the second file::
+
+     '--src', '/path/to/file1.g3', '/path/to/file2.g3',
+
+You can also tell the magpie agent to ignore the src argument all together and generate
+fake data by adding the ``--fake-data`` argument.
+
+
+Docker Compose
+``````````````````
+
+Below is an example docker-compose entry for running a magpie corresponding to
+crate-id 1 and slot 2. If crossbar is being run on a different server, you'll
+have to modify the ``site-hub`` and ``site-http`` args accordingly::
+
+    ocs-magpie-crate1slot2:
+        image: simonsobs/ocs-magpie:${SOCS_TAG}
+        hostname: ocs-docker
+        user: ocs:ocs
+        network_mode: host
+        container_name: ocs-magpie
```

Comment:
Is there a reason to name the container here? (i.e. something is relying on that name to resolve) If not I think this should be removed to avoid conflicts when running multiple instances for different crates/slots. (Or the name here made more specific, like `ocs-magpie-crate1slot2`, but letting docker-compose do the naming should be fine too.)

Reply:
no good reason other than convenience. I'll remove